futurenhs.api/FutureNHS.Api/Services/FileService.cs

```diff
@@ -171,6 +171,7 @@ private async Task<FileDto> UploadMultipartContent(Stream requestBody, string? c
                             catch (Exception ex)
                             {
                                 _logger.LogError(ex, "An error occurred uploading file to blob storage");
+                                await _blobStorageProvider.DeleteFileAsync(uniqueFileName);
```

Comment:
There is a chance this can fail and throw an exception. Could we get some further logging along the lines of "An error occurred while uploading the file {uniqueFileName} to blob storage, and it was not possible to delete the file"

Or something like that, so we can have an indication of orphaned files in the system.